backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -954,6 +954,39 @@ trait StandardAsyncExecutionActor
     }
   }
 
+  /**
+   * Merge key-value pairs from previous job execution attempt with incoming pairs from backend and store them in
+   * the database. In case when there are key-value pairs with the same key name among those from previous attempt and
+   * coming from backend, backend values have higher precedence.
+   * `FailedRetryCountKey` is handled separately outside of this method 
+   *
+   */
+  private def saveKvPairsForNextAttempt(backendExecutionStatus: Future[ExecutionHandle]): Future[ExecutionHandle] = {
```

Comment:
I think I would:
* Change this method signature to `saveKvPairsForNextAttempt(backendExecutionStatus: ExecutionHandle): Future[ExecutionHandle]`
* Refactor the `flatMap` above into a for-comprehension over a chain of `Future[X]` methods.

UPDATE: I think actually this comment might be redundant given my next one...